forest_lib/forest/aws.py

```diff
@@ -0,0 +1,194 @@
+        :param file_name: base name of file to be queried
+        :returns: logical indicating file existence
+        """
+        return self.remote_file_exists(file_name)
+
+    def remote_file_exists(self, file_name):
+        """Check file present in S3 bucket
+
+        :param file_name: base name of file to be queried
+        :returns: logical indicating file existence
+        """
+        return self.s3_file_exists(self.s3_url(file_name))
+
+    @staticmethod
+    def s3_file_exists(url):
+        """Check whether a remote file exists; return Bool.
+
+        Check whether a file at the remote location specified by remore
+        path exists by trying to open a url request.
+
+        :param remote_path: Path to check for file at.
+        :returns: logical indicating file existence
+        """
+        try:
+            _ = urllib.request.urlopen(remote_path)
+            print('file {0} found at remote location.'.format(remote_path))
+            return True
+        except urllib.error.HTTPError:
+            warning_msg1 = 'warning: file {0} NOT found at remote location.'
+            warning_msg1 = warning_msg1.format(remote_path)
+            print(warning_msg1)
+        return False
+
+    def local_file_exists(self, file_name):
+        """Check file present on disk
+
+        :param file_name: base name of file
+        :returns: logical indicating file existence
+        """
+        return os.path.isfile(self.path_to_load(file_name))
+
+    def path_to_load(self, file_name):
+        """Compute string representation of downloaded file
+
+        :param file_name: base name of file
+        :returns: full path of file
+        """
+        return os.path.join(self.download_directory, file_name)
+
+    def load_file(self, file_name):
+        """Do download from S3 bucket if file not already on disk
+
+        :param file_name: base name of file
+        :returns: path_to_file
+        """
+        if not os.path.isdir(self.download_directory):
+            print("creating directory {0}".format(self.download_directory))
+            os.makedirs(self.download_directory)
+        if not self.local_file_exists(file_name) and self.remote_file_exists(file_name):
+            self.s3_download(self.s3_url(file_name),
+                             self.path_to_load(file_name))
+        return self.path_to_load(file_name)
+
+    def s3_url(self, file_name):
+        return os.path.join(self.server_address,
+                            self.bucket_name,
+                            "model_data",
+                            file_name)
+
+    @staticmethod
+    def s3_download(url, path):
+        """port of forest.util.download_from_s3"""
+        if not os.path.isfile(path):
+            print('retrieving file from {0}'.format(url))
+            urllib.request.urlretrieve(url, path)
+            print('file {0} downloaded'.format(path))
+        else:
+            print(path, ' - File already downloaded')
+
+
+class S3Mount(object):
```

Comment:
I don't feel anything about this is intrinsically S3. It's just a local mount/disk/directory. The S3 nature is an implementation detail that should be discussed in the read me but if we FUSE mount Azure or Google object store this class still works.